### ISSUE TITLE
Build failure (current rev. f03a896):  undefined reference to `pow', etc

### ISSUE THREAD

**Issue by yurivict** [psi4/psi4#853]
FreeBSD 11.1, clang-40
Obviously, ```-lm``` is missing somewhere.

```
libxc.a(gga_x_bayesian.c.o): In function `work_gga_x':
/usr/ports/science/psi4/work/.build/external/upstream/libxc/libxc_external-prefix/src/libxc_external/src/gga_x_bayesian.c:(.text+0x583): undefined reference to `pow'
libxc.a(gga_x_q2d.c.o): In function `work_gga_x':
/usr/ports/science/psi4/work/.build/external/upstream/libxc/libxc_external-prefix/src/libxc_external/src/gga_x_q2d.c:(.text+0x8d1): undefined reference to `pow'
libxc.a(gga_c_am05.c.o): In function `work_gga_c':
/usr/ports/science/psi4/work/.build/external/upstream/libxc/libxc_external-prefix/src/libxc_external/src/gga_c_am05.c:(.text+0x1d3): undefined reference to `pow'
/usr/ports/science/psi4/work/.build/external/upstream/libxc/libxc_external-prefix/src/libxc_external/src/gga_c_am05.c:(.text+0x24d): undefined reference to `pow'
```

**Comment by loriab** [comment]
From the little I know about FreeBSD, I'm guessing that it's not fitting into either the Mac or Linux channels that our build system expects. The build is well-tested by devs with Mac+AppleClang and Linux+Intel(GCC), and CI covers Linux+GCC and Linux+Clang. It's going to be tricky to debug remotely, but if you work it out and care to submit a PR, we'd be grateful (particularly if you updated to master from 1.1).

One thing to watch out for (wrt #852) is to make sure CMake can't find a different (higher than 2.0) version of pybind11 _anywhere_ on your system. We try to nail it down in CMake, but sometimes it finds and uses a wrong version installed alongside another dependency.

Naturally, if I recognize the cause for an error, I'm glad to help.

**Comment by yurivict** [comment]
> From the little I know about FreeBSD, I'm guessing that it's not fitting into either the Mac or Linux channels that our build system expects. 

It's somewhere in between.

**Comment by yurivict** [comment]
There are also warnings with clang-40:
```
In file included from /usr/ports/science/psi4/work/psi4-f03a896/psi4/src/psi4/libfock/PK_workers.cc:32:
/usr/ports/science/psi4/work/psi4-f03a896/psi4/src/psi4/libfock/PK_workers.h:254:46: warning: control reaches end of non-void function [-Wreturn-type]
    PKWorker & operator = (PKWorker &other) {}
                                             ^
In file included from /usr/ports/science/psi4/work/psi4-f03a896/psi4/src/psi4/libfock/PKmanagers.cc:25:
/usr/ports/science/psi4/work/psi4-f03a896/psi4/src/psi4/libfock/PK_workers.h:254:46: warning: control reaches end of non-void function [-Wreturn-type]
    PKWorker & operator = (PKWorker &other) {}
                                             ^
In file included from /usr/ports/science/psi4/work/psi4-f03a896/psi4/src/psi4/libfock/apps.cc:32:
/usr/ports/science/psi4/work/psi4-f03a896/psi4/src/psi4/libfock/hamiltonian.h:402:84: warning: control reaches end of non-void function [-Wreturn-type]
            const std::pair<std::shared_ptr<Vector>, std::shared_ptr<Vector> >& x){} ;
                                                                                   ^
/usr/ports/science/psi4/work/.build/external/upstream/libxc/libxc_external-prefix/src/libxc_external/src/work_gga_c.c:86:5: warning: implicit declaration of function 'func' is invalid in C99 [-Wimplicit-function-declaration]
    func(p, &r);
    ^
/usr/ports/science/psi4/work/.build/external/upstream/libxc/libxc_external-prefix/src/libxc_external-build/src/work_gga_x.c:152:7: warning: implicit declaration of function 'func' is invalid in C99 [-Wimplicit-function-declaration]
      func(p, order, x, &f, &dfdx, &d2fdx2, &d3fdx3);
      ^
```

**Comment by loriab** [comment]
For the most part, we let CMake figure out `-lm`. We'll try to catch the clang warnings in a cleanup pass. If there's anything you'd like to submit a PR for, please do so. The dev community doesn't have a FreeBSD machine accessible. Thanks for packaging!
